woocommerce/woocommerce-gateway-stripe
# Load xdebug on demand for VSCode and Docker

## Changes proposed in this Pull Request:

I noticed that the dev container was extremely slow, especially on Linux. After some investigation, I figured out it was because xdebug was enabled for every request, even if we were not using it.

This PR adds two tasks in the vscode debug config file to enable xdebug only when the debugger is started and disable it when it is stopped. It also updates the xdebug version in the Dockerfile to version 3.

- Update xdebug version on dev docker container to version 3
- Update Dockerfile to leave xdebug disabled by default
- Update vscode config file to enable and disable xdebug on demand

## Testing instructions

Please test if the debug is running without additional configurations needed:

1. Run `npm run down` to stop your dev environment if it's running.
2. Run `docker-compose build --no-cache` to rebuild the development image
3. Run `npm run up` to start the containers.
4. Open the Visual Studio Code in the project root folder
5. Add a breakpoint in the main plugin file
6. Start the debugger
7. Visit the dev site on [http://localhost:8082/](http://localhost:8082/)
8. The debugger should stop in your breakpoint.
9. Stop the debugger.

If the debugger is working, everything is good. The speed gain is more noticeable on Linux. 

---


**Post merge**
